### PR TITLE
CAF-2380: Getting started guide documentation re-write for Audit 3.0.0 release

### PR DESCRIPTION
**Pull request by CJComac**

### CODE REVIEW

File: docs/pages/en-us/Getting-Started.md [hunk 1]
```diff
@@ -13,185 +13,33 @@ banner:
...
+You need to perform the following steps to set up the Audit service.
 
-1. Install and configure HPE Vertica 7.2.x on a dedicated machine.
-2. Install and configure Apache Kafka on a dedicated machine.
-3. Using the Chateau deployment toolset, launch the Apache Kafka-Vertica scheduler and the Audit Management web service in Mesos/Marathon.
-4. Define an application's audit events in an Audit Event Definition File.
-5. Register the audit event definition file and add tenant(s) with the Audit Management web service. This creates the necessary database tables in Vertica.
-6. Generate the client-side auditing library using the audit event definition file and the code generation plugin. 
-7. In your application, use the client-side auditing library to send audit events to Kafka. The Audit Scheduler will automatically load the events from Kafka into HPE Vertica.
+1. Using Audit Service Deployment, launch an Elasticsearch cluster.
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: Use the [Audit Service Deploy Project](https://github.com/CAFAudit/audit-service-deploy), to launch an Elasticsearch cluster.

File: docs/pages/en-us/Getting-Started.md [hunk 2]
```diff
@@ -13,185 +13,33 @@ banner:
...
-- [Deploying Kafka](#deploying-kafka)
-- [Deploying Audit Web Service and Kafka-Vertica Scheduler](#deploying-audit-web-service-and-kafka-vertica-scheduler)
+- [Deploying Elasticsearch](#deploying-elasticsearch)
 - [Writing an Application Audit Event Definition File](#writing-an-application-audit-event-definition-file)
-- [Using the Audit Management Web Service](#using-the-audit-management-web-service)
 - [Generating a Client-side Auditing Library](#generating-a-client-side-auditing-library)
 - [Using the Client-side Auditing Library](#using-the-client-side-auditing-library)
 
-For more information on Chateau, go [here](https://github.hpe.com/caf/chateau).
+For more information on Audit Service Deployment, go [here](https://github.com/CAFAudit/audit-service-deploy).
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: For more information on Audit Service Deploy, go [here](https://github.com/CAFAudit/audit-service-deploy).

File: docs/pages/en-us/Getting-Started.md [hunk 3]
```diff
@@ -13,185 +13,33 @@ banner:
...
-- For enterprise deployments of HPE Vertica, add the details of your host and user accounts to Chateau's environment/vertica.json file.
-
-- For enterprise deployments of Kafka, add the details of your Kafka brokers to Chateau's environment/kafka.json file.
-
-- To deploy the Audit Management web service and the Kafka-Vertica Scheduler, follow the [Service Deployment](https://github.hpe.com/caf/chateau/blob/develop/deployment.md) guide and use the following option with the deployment shell script: `./deploy-service.sh audit`
-
-The following figure shows a Marathon environment running the Audit services started with Chateau:
-
-![Marathon running CAF Audit Management Service](images/MarathonWithAuditManagementService.PNG)
+For developer deployment of Elasticsearch, please follow the Audit Service Deployment documentation as it covers configuring and starting of Elasticsearch cluster in Docker. For more information on Audit Service Deployment, go [here](https://github.com/CAFAudit/audit-service-deploy).
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: For a developer deployment of Elasticsearch, please follow the Audit Service Deployment documentation.  The documentation covers configuring and starting of an Elasticsearch cluster in Docker.

File: docs/pages/en-us/Getting-Started.md [hunk 4]
```diff
@@ -218,7 +66,7 @@ If you reference the XML schema file from your audit event definition file, then
 
 	<AuditedApplication xmlns="http://www.hpe.com/CAF/Auditing/Schema/AuditedApplication.xsd"
 	                    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	                    xsi:schemaLocation="http://www.hpe.com/CAF/Auditing/Schema/AuditedApplication.xsd http://rh7-artifactory.svs.hpeswlab.net:8081/artifactory/policyengine-release/com/hpe/caf/caf-audit-schema/1.1/caf-audit-schema-1.1.jar!/schema/AuditedApplication.xsd">	                   
+	                    xsi:schemaLocation="http://www.hpe.com/CAF/Auditing/Schema/AuditedApplication.xsd https://oss.sonatype.org/content/repositories/releases/com/github/cafaudit/caf-audit-schema/3.0.0/caf-audit-schema-3.0.0.jar!/schema/AuditedApplication.xsd">	                   
 
 Many IDEs and XML editors use the schema file to provide IntelliSense and type-ahead when authoring the definition file.
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: Many IDEs and XML editors use the schema file to provide IntelliSense / Auto-Complete when authoring the definition file.

File: docs/pages/en-us/Getting-Started.md [hunk 5]
```diff
@@ -266,53 +114,9 @@ The following is an example of an audit event definition file used throughout th
...
-
-![CAF Audit Account 1 Kafka Reject Table Columns](images/account_1RejectTable.png)
-
 ## Generating a Client-side Auditing Library
 
-As previously mentioned, in order to use Auditing, you must first define the audit events in an audit event definition file. After you create the definition file, you use it to generate a client-side library. Technically, you do not need to generate a client-side library to use Auditing; you could use the `caf-audit` module directly. However, generating a client-side library should make it easier and safer to raise events because each event can be raised with a single type-safe call.
+As previously mentioned, in order to use Auditing, you must first define the audit events in an audit event definition file. After you create the definition file, you use it to generate a client-side library. Technically, you do not need to generate a client-side library to use Auditing; you could use the `caf-audit` module directly; read more on the using the library directed [here](Client-API). However, generating a client-side library should make it easier and safer to raise events because each event can be raised with a single type-safe call.
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: library directed should be "library directly, [here](Client-API)."

File: docs/pages/en-us/Getting-Started.md [hunk 6]
```diff
@@ -440,14 +228,14 @@ A generated client-side library should be referenced in the normal way in the ap
...
-	    <version>1.0-SNAPSHOT</version>
+	    <version>1.0.0-SNAPSHOT</version>
 	</dependency>
 
 ### Audit Connection
 
 Regardless of whether you choose to use a generated client-side library, or to use `caf-audit` directly, you must first create an `AuditConnection` object.
 
-This object represents a logical connection to persistent storage (that is, Kafka in the current implementation). It is a thread-safe object. ***Please take into account that this object requires some time to construct. The application should hold on to it and re-use it, rather than constantly re-construct it.***
+This object represents a logical connection to persistent storage (that is, Elasticsearch in the current implementation). It is a thread-safe object. ***Please take into account that this object requires some time to construct. The application should hold on to it and re-use it, rather than constantly re-construct it.***
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: This object represents a logical connection to persistent storage  
Update to:  
This object represents a logical connection to the datastore

File: docs/pages/en-us/Getting-Started.md [hunk 7]
```diff
@@ -539,33 +327,35 @@ In the `ConfigurationSource` above, we used JSON-encoded files with the followin
...
 - `CAF_APPNAME: sampleappgroup/sampleapp`
 
-Given this configuration, you would configure Auditing by creating a file named `cfg_sampleappgroup_sampleapp_KafkaAuditConfiguration` in the `/etc/sampleapp/config/` directory. The contents of this file should be similar to the following:
+Given this configuration, you would configure Auditing by creating a file named `cfg_sampleappgroup_sampleapp_ElasticsearchAuditConfiguration` in the `/etc/sampleapp/config/` directory. The contents of this file should be similar to the following:
 
 	{
-	    "bootstrapServers": "<kafka broker>:<port number>",
-	    "acks": "all",
-	    "retries": "0"
+	    "hostAndPort": "<Elasticsearch_Cluser_Node_1>:<Port_Number>,<Elasticsearch_Cluser_Node_2>:<Port_Number>",
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: Side note:

The parameter hostAndPort is going to change to "hostAndPortValues".

This change will hopefully be made tomorrow, so feel free to update the documentation in advance.

File: docs/pages/en-us/Getting-Started.md [hunk 8]
```diff
@@ -539,33 +327,35 @@ In the `ConfigurationSource` above, we used JSON-encoded files with the followin
...
+	    "numberOfShards": "5",
+	    "numberOfReplicas": "1"
 	}
 
 where:
 
-- `bootstrapServers` refers to one or more of the nodes of the Kafka cluster as a comma-separated list.
-- `acks` is the number of nodes in the cluster which must acknowledge an audit event when it is sent.
-- `retries` is the number of additional attempts in case of issues communicating with the Kafka cluster.
+- `hostAndPort` refers to one or more of the nodes of the Elasticsearch cluster as a comma-separated list.
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: hostAndPort to become hostAndPortValues

File: docs/pages/en-us/Getting-Started.md [hunk 9]
```diff
@@ -539,33 +327,35 @@ In the `ConfigurationSource` above, we used JSON-encoded files with the followin
...
+- `numberOfShards` the number of primary shards that an index should have. Defaults to 5.
+- `numberOfReplicas` the number of replica shards (copies) that each primary shard should have. Defaults to 1.
 
 ### Audit Channel
 
 After you successfully construct an `AuditConnection` object, you must construct an `AuditChannel` object.
 
-This object represents a logical channel to the persistent storage (that is, Kafka in this implementation). **It is NOT a thread-safe object and must not be shared across threads without synchronization.** However, you will have no issue constructing multiple `AuditChannel` objects simultaneously on different threads. The objects are lightweight and caching them is not that important.
+This object represents a logical channel to the persistent storage (that is, Elasticsearch in this implementation). **It is NOT a thread-safe object and must not be shared across threads without synchronization.** However, you will have no issue constructing multiple `AuditChannel` objects simultaneously on different threads. The objects are lightweight and caching them is not that important.
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: persistent storage --> datastore

File: docs/pages/en-us/Getting-Started.md [hunk 10]
```diff
@@ -539,33 +327,35 @@ In the `ConfigurationSource` above, we used JSON-encoded files with the followin
...
 
 The `AuditChannel` object can be constructed using the `createChannel()` method on the `AuditConnectionn` object. It does not take any parameters.
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: Typo: `AuditConnectionn` should be `AuditConnection`

File: docs/pages/en-us/Getting-Started.md [hunk 11]
```diff
@@ -598,12 +388,18 @@ Following is an example for a SampleApp's `viewDocument` event, which takes a si
 
 The name of the event is included in the generated method name. In addition to the custom parameters (document id in this case), the caller must pass the `AuditChannel` object to be used, as well as the tenant id, user id, and correlation id.
 
-The method will throw an exception if the audit event could not be stored for some reason (for example, network failure).
+The method will throw an exception if the audit event could not be stored for some reason (for example, network failure or supplying unsupported tenantId characters).
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md:393]: Are we now offering some form of validation on the tenantId?  If so great, if not we shouldn't suggest that we are in the document.

File: docs/pages/en-us/Getting-Started.md [hunk 12]
```diff
@@ -598,12 +388,18 @@ Following is an example for a SampleApp's `viewDocument` event, which takes a si
...
 ### Verification Instructions
 
-Every time an `AuditLog` method is called, a new row is entered into the tenant's audit application table. The following figure shows a tenant's `account_1` schema's `AuditSampleApp` table with an audit event entry with data for an audit event:
+Every time an `AuditLog` method is called, a new application audit event is entered into the tenant's index. Elasticsearch offers a RESTful interface for querying index entries.
+
+The following figure shows all audit event entries belonging to tenant index `00000001_audit`:
+
+![Tenant 00000001_audit all audit event index entries](images/GetTenantIndexAuditEvents.PNG)
+
+The following figure shows all audit event entries belonging to tenant index `00000001_audit` whose `userId` is `JoanneBloggs@yourcompany.com`:
```
Comment (glucy) [inline docs/pages/en-us/Getting-Started.md]: Is Joe on his holidays?  :-P

**Review by glucy — CHANGES_REQUESTED** [review]
Further reviewing required once a developer build has been created.

File: docs/pages/en-us/Getting-Started.md (hunk 11 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md:393]: We currently throw an illegal argument exception if you try to pass commas within the tenantId

File: docs/pages/en-us/Getting-Started.md (hunk 12 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Joanne might be able to answer that :)

File: docs/pages/en-us/Getting-Started.md (hunk 10 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Corrected in the latest commit

File: docs/pages/en-us/Getting-Started.md (hunk 9 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Adjusted in the latest commit

File: docs/pages/en-us/Getting-Started.md (hunk 8 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Adjusted in the latest commit

File: docs/pages/en-us/Getting-Started.md (hunk 7 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Adjusted in the latest commit

File: docs/pages/en-us/Getting-Started.md (hunk 6 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Adjusted in the latest commit

Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Adjusted in the latest commit

File: docs/pages/en-us/Getting-Started.md (hunk 5 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Corrected in the latest commit

File: docs/pages/en-us/Getting-Started.md (hunk 4 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Changed in the latest commit

File: docs/pages/en-us/Getting-Started.md (hunk 3 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Changed in the latest commit

File: docs/pages/en-us/Getting-Started.md (hunk 2 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Changed in the latest commit

File: docs/pages/en-us/Getting-Started.md (hunk 1 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Changed in the latest commit

File: docs/pages/en-us/Getting-Started.md [hunk 13]
```diff
@@ -539,33 +327,35 @@ In the `ConfigurationSource` above, we used JSON-encoded files with the followin
 - `CAF_CONFIG_PATH: /etc/sampleapp/config`
 - `CAF_APPNAME: sampleappgroup/sampleapp`
 
-Given this configuration, you would configure Auditing by creating a file named `cfg_sampleappgroup_sampleapp_KafkaAuditConfiguration` in the `/etc/sampleapp/config/` directory. The contents of this file should be similar to the following:
+Given this configuration, you would configure Auditing by creating a file named `cfg_sampleappgroup_sampleapp_ElasticsearchAuditConfiguration` in the `/etc/sampleapp/config/` directory. The contents of this file should be similar to the following:
```
Comment (mulhollandc) [inline docs/pages/en-us/Getting-Started.md]: I don't think this is correct. The configuration file I think needs to match the configuration class name, which in this case is actually ElasticAuditConfiguration.

File: docs/pages/en-us/Getting-Started.md [hunk 14]
```diff
@@ -539,33 +327,35 @@ In the `ConfigurationSource` above, we used JSON-encoded files with the followin
...
 
 	{
-	    "bootstrapServers": "<kafka broker>:<port number>",
-	    "acks": "all",
-	    "retries": "0"
+	    "hostAndPortValues": "<Elasticsearch_Cluser_Node_1>:<Port_Number>,<Elasticsearch_Cluser_Node_2>:<Port_Number>",
```
Comment (mulhollandc) [inline docs/pages/en-us/Getting-Started.md]: Just to note that hostAndPortValues probably won;t work right now. caf-audit currently expects hostAndPort.

File: docs/pages/en-us/Getting-Started.md [hunk 15]
```diff
@@ -539,33 +327,35 @@ In the `ConfigurationSource` above, we used JSON-encoded files with the followin
...
+	    "numberOfShards": "5",
+	    "numberOfReplicas": "1"
 	}
 
 where:
 
-- `bootstrapServers` refers to one or more of the nodes of the Kafka cluster as a comma-separated list.
-- `acks` is the number of nodes in the cluster which must acknowledge an audit event when it is sent.
-- `retries` is the number of additional attempts in case of issues communicating with the Kafka cluster.
+- `hostAndPortValues` refers to one or more of the nodes of the Elasticsearch cluster as a comma-separated list.
```
Comment (mulhollandc) [inline docs/pages/en-us/Getting-Started.md:341]: Same as above for hostAndPortValues.

File: docs/pages/en-us/Getting-Started.md [hunk 16]
```diff
@@ -539,33 +327,35 @@ In the `ConfigurationSource` above, we used JSON-encoded files with the followin
...
+- `clusterName` name of the Elasticsearch cluster. Defaults to "elasticsearch".
+- `numberOfShards` the number of primary shards that an index should have. Defaults to 5.
+- `numberOfReplicas` the number of replica shards (copies) that each primary shard should have. Defaults to 1.
 
 ### Audit Channel
 
 After you successfully construct an `AuditConnection` object, you must construct an `AuditChannel` object.
```
Comment (mulhollandc) [inline docs/pages/en-us/Getting-Started.md]: I'd re-word to "After you successfully construct an `AuditConnection` object, you must then create ..."

File: docs/pages/en-us/Getting-Started.md [hunk 17]
```diff
@@ -598,12 +388,18 @@ Following is an example for a SampleApp's `viewDocument` event, which takes a si
...
 
-The method will throw an exception if the audit event could not be stored for some reason (for example, network failure).
+The method will throw an exception if the audit event could not be stored for some reason (for example, network failure or supplying unsupported tenantId characters).
 
 ### Verification Instructions
 
-Every time an `AuditLog` method is called, a new row is entered into the tenant's audit application table. The following figure shows a tenant's `account_1` schema's `AuditSampleApp` table with an audit event entry with data for an audit event:
+Every time an `AuditLog` method is called, a new application audit event is entered into the tenant's index. Elasticsearch offers a RESTful interface for querying index entries.
+
+The following figure shows all audit event entries belonging to tenant index `00000001_audit`:
```
Comment (mulhollandc) [inline docs/pages/en-us/Getting-Started.md]: Explain how you got the results. Just a URL in the browser? If so, what format is expected?

File: docs/pages/en-us/Getting-Started.md (hunk 12 above)
Comment (mulhollandc) [inline docs/pages/en-us/Getting-Started.md]: Same comment as above for other figure.

**Review by mulhollandc — CHANGES_REQUESTED** [review]
Final comment would be have you tested the actual code snippets that you are including on this page to make sure that when it is all brought together that an audit event message is successfully sent and indexed into Elasticsearch?

File: docs/pages/en-us/Getting-Started.md (hunk 14 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: 3.0.0 will be released as hostAndPortValues. Given the current snapshot, during my manual test I will need to change this

File: docs/pages/en-us/Getting-Started.md (hunk 15 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md:341]: 3.0.0 will be released as hostAndPortValues. Given the current snapshot, during my manual test I will need to change this

File: docs/pages/en-us/Getting-Started.md (hunk 12 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: In the latest commit, reworded to:
"The following Search API operation, `http://<Elasticsearch_Cluser_Node_1>:<Port_Number>/00000001_audit/cafAuditEvent/_search?q=userId:JoanneBloggs@yourcompany.com`, can be run to display the audit event entries belonging to tenantId `00000001` and whose `userId` is `JoanneBloggs@yourcompany.com`:"

File: docs/pages/en-us/Getting-Started.md (hunk 17 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: In the latest commit, reworded to:
"Every time an `AuditLog` method is called, a new application audit event is entered into the tenant's index. 

Elasticsearch offers a RESTful interface for querying index entries. For more information on the Elasticsearch REST Search API, go [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/search.html).

Using the configuration source details above, the following Search API operation, `http://<Elasticsearch_Cluser_Node_1>:<Port_Number>/00000001_audit/cafAuditEvent/_search`, can be run to display all audit event entries belonging to tenantId `00000001`:"

File: docs/pages/en-us/Getting-Started.md (hunk 16 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Adjusted in the latest commit

File: docs/pages/en-us/Getting-Started.md (hunk 13 above)
Comment (CJComac) [reply docs/pages/en-us/Getting-Started.md]: Adjusted in the latest commit

**Review by mulhollandc — APPROVED** [review]

**Review by glucy — APPROVED** [review]

**Pull request merged**